framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -265,86 +267,124 @@ def checkConvergence(self):
         break
     return convergence
 
-  def _updateConvergenceVector(self, traj, varsUpdate, currentLossValue):
+  def _updateConvergenceVector(self, traj, varsUpdate, currentLossVal):
     """
       Local method to update convergence vector.
       @ In, traj, int, identifier of the trajector to update
       @ In, varsUpdate, int, current variables update iteration number
-      @ In, currentLossValue, float, current loss function value
+      @ In, currentLossVal, float, current loss function value
       @ Out, None
     """
-    if not self.convergeTraj[traj]:
-      if len(self.counter['gradientHistory'][traj][1]) < 1:
-        self.status[traj]['reason'] = 'found new opt point'
-      else:
-        objectiveOutputs = np.zeros(self.gradDict['numIterForAve'])
-        for i in range(self.gradDict['numIterForAve']): #evens are opt point evaluations
-          index = i*2
-          objectiveOutputs[i] = self.getLossFunctionGivenId(self._createEvaluationIdentifier(traj,varsUpdate-1,index))
-        if any(np.isnan(objectiveOutputs)):
-          self.raiseAnError(Exception,"the objective function evaluation for trajectory " +str(traj)+ "and iteration "+str(varsUpdate-1)+" has not been found!")
-        oldVal = objectiveOutputs.mean() # TODO should this be from counter[recentOptPoints][traj]?
-        # see if new point is better than old point
-        newerIsBetter = self.checkIfBetter(currentLossValue,oldVal)
-        varK = self.denormalizeData(self.optVarsHist[traj][self.counter['varsUpdate'][traj]])
-        ## convergence values
-        # gradient norm
-        gradNorm  = self.counter['gradNormHistory'][traj][0]
+    # if trajectory is already converged, no need to update. #FIXME then why was this method called?
+    if self.convergeTraj[traj]:
+      return
```

Comment:
I'm running the coverage on this branch to check that.  I'd be surprised if it does, but I wasn't confident enough to remove it during development.